src/Impostor.Server/Net/State/Game.Data.cs

```diff
@@ -239,7 +241,17 @@ public async ValueTask<bool> HandleGameDataAsync(IMessageReader parent, ClientPl
                             return false;
                         }
 
-                        sender.Scene = reader.ReadString();
+                        // According to game assembly, sender is only allowed to send OnlineGame.
+                        if (scene != "OnlineGame")
+                        {
+                            _logger.LogWarning(
```

Comment:
I think this should be an anticheat check: although there are very few reasons for protocol violations like this, I don't we ever straight up block packets

Reply:
I agree @miniduikboot. Also why were we never reading the scene before?

Reply:
I don't know, nobody has needed to touch this part of the code in the past 4 years, and I'm not actively hunting through vulnerabilities in the code

Reply:
What is the best category for anticheat kick?

Reply:
I'd say GameFlow but considering the not important crap that's already in there I'm not so sure anymore.

Also considering that hosts get permitted to break anticheat rules on most servers via the AllowCheatingHosts configuration option, we could consider making a separate category that overrules that option. I don't see a practical reason for these exploits other than research

Reply:
Btw, this code is meant to work exactly like IS servers, even in modded protocol, the code is still cancelled by game servers.

Reply:
We usually try to follow what Innersloth does, but we deviate every now and then. Considering the result on players I'm in favor of merging this patch, but I'm not sure yet if there should be an opt-out

Reply:
I mean, no mods will ever use scene change in this way, so in my opnion its better to always handle it, considering its a potential cheat.

Reply:
Should be fine to merge like this then. Seems like there is no reason not to.